controllers/restic.go

```diff
@@ -149,16 +164,25 @@ func (r *DPAReconciler) buildResticDaemonset(dpa *oadpv1alpha1.DataProtectionApp
 		install.WithAnnotations(dpa.Spec.PodAnnotations),
 		install.WithSecret(false))
 	// Update Items in ObjectMeta
+	dsName := ds.Name
 	ds.TypeMeta = installDs.TypeMeta
 	// Update Spec
 	ds.Spec = installDs.Spec
-	ds.Labels = installDs.Labels
+	ds.ObjectMeta = installDs.ObjectMeta
+	ds.Name = dsName
 
 	return r.customizeResticDaemonset(dpa, ds)
 }
 
 func (r *DPAReconciler) customizeResticDaemonset(dpa *oadpv1alpha1.DataProtectionApplication, ds *appsv1.DaemonSet) (*appsv1.DaemonSet, error) {
-
+	// add custom pod labels
+	if dpa.Spec.Configuration.Restic != nil && dpa.Spec.Configuration.Restic.PodConfig != nil && dpa.Spec.Configuration.Restic.PodConfig.Labels != nil {
```

Comment:
Just a NIT to maybe clean this up. If `dpa.Spec.Configuration.Restic` is `nil`, should we be returning from this func?

Reply:
Should not return yet becauase we are still customizing restic daemonset after this condition exits.